### PR TITLE
RIIR update_lints: use WalkDir instead of read_dir

### PR DESCRIPTION
**Pull request by phansch**
We need to traverse into subdirectories in order to collect all lints. `read_dir` doesn't do that and `WalkDir` seems like the safest/quickest way to achieve that.

### CODE REVIEW

**Comment by phansch** [comment]
(I was super surprised that `WalkDir` was a straightforward drop-in replacement!)

File: clippy_dev/src/lib.rs [hunk 1]
```diff
@@ -89,9 +92,9 @@ fn parse_contents(content: &str, filename: &str) -> impl Iterator<Item=Lint> {
 }
 
 /// Collects all .rs files in the `clippy_lints/src` directory
-fn lint_files() -> impl Iterator<Item=fs::DirEntry> {
-    fs::read_dir("../clippy_lints/src")
-        .unwrap()
+fn lint_files() -> impl Iterator<Item=walkdir::DirEntry> {
+    WalkDir::new("../clippy_lints/src")
```
Comment (oli-obk) [inline clippy_dev/src/lib.rs]: can you add a comment here that we use `WalkDir` instead of `fs::read_dir` to recurse into the directories?

**Comment by oli-obk** [comment]
r=me with nit fixed

File: clippy_dev/src/lib.rs (hunk 1 above)
Comment (phansch) [reply clippy_dev/src/lib.rs]: yup!

**Comment by phansch** [comment]
going to hold off on merging due to https://github.com/hyperium/mime/issues/93

**Comment by mati865** [comment]
@phansch `mime` fix was released just 17 minutes after your comment :smile:

**Pull request merged**